### PR TITLE
Introduce RestAssured to test checkout

### PR DESCRIPTION
**Pull request by aguairon**
blah blah blah some useful stuff about what I did.

### CODE REVIEW

File: src/test/java/api/SiteApi.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public SiteApi() {
+        this("https://www.treatwell.co.uk", "api");
+    }
+
+    public SiteApi(String baseURI, String basePath) {
+        RestAssured.baseURI = baseURI;
+        RestAssured.basePath = basePath;
+    }
+
+
```
Comment (gregbeech) [inline src/test/java/api/SiteApi.java]: Hmm, looks like you've got two lines spacing between these functions. The Java standard is to have only one.

File: src/test/java/api/SiteApi.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public String post(String body, String specificPath, int statusCode) {
+         return given().contentType("application/json").body(body).
+                 when().post(specificPath).
+                 then().statusCode(statusCode)
+                 .extract().response().asString();
```
Comment (gregbeech) [inline src/test/java/api/SiteApi.java]: This formatting is a bit inconsistent; on line 20 you ended with a `.` but here you're starting with it.

File: src/test/java/api/SiteApi.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public void close() throws Exception {
+
+    }
```
Comment (gregbeech) [inline src/test/java/api/SiteApi.java:22]: Should this do something?

**Review by gregbeech — CHANGES_REQUESTED** [review]
Good start, but some things that could be improved.

File: src/test/java/api/SiteApi.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        RestAssured.baseURI = baseURI;
+        RestAssured.basePath = basePath;
+    }
+
+
+    public String post(String body, String specificPath, int statusCode) {
+         return given().contentType("application/json").body(body).
+                 when().post(specificPath).
+                 then().statusCode(statusCode).
+                 extract().response().asString();
```
Comment (gregbeech) [inline src/test/java/api/SiteApi.java:21]: This code is indented differently to how I'd expect. When chaining functions over multiple lines, acceptable indentation is either all the lines with the same starting column, e.g.

```java
return line1.
       line2.
       line3;
```

...or all the lines indented by 4, e.g.

```java
return 
    line1.
    line2.
    line3;
```

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+
+//    @Before
+//    public void navigateToCheckoutPage() {
+//        browsePage = site.navigateToBrowsePage();
+//        browsePage.closeCookieBanner();
+//        offerForm = browsePage.selectFirstAvailableOffer();
+//        availabilityPage = offerForm.selectSkuAndOpenAvailabilityPage();
+//        checkoutPage = availabilityPage.goToCheckout();
+//    }
```
Comment (gregbeech) [inline src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:31]: Should this commented out code be here?

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+public class CheckoutSubmissionTest {
+    WebSite site = new WebSite();
+    BrowsePage browsePage;
+    OfferForm offerForm;
+    AvailabilityPage availabilityPage;
+    CheckoutPage checkoutPage;
+    CardFactory cardFactory = new CardFactory();
+    Card card = cardFactory.build();
+    SiteApi siteApi = new SiteApi();
+    CheckoutApi checkoutApi = new CheckoutApi();
```
Comment (gregbeech) [inline src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:21]: Looks like quite a few of these things aren't used? Also generally it's good practice to specify `private` if they're only intended for use inside the class, and `final` if they're not intended to be reassigned.

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
...317696}],\"venueId\":314127,\"recipients\":[{\"name\":\"etwtwtwe\"}]}],\"startTime\":\"825\",\"date\":\"2017-01-16\",\"payment\":{\"advisoryOrderTotal\":\"3\",\"paymentType\":\"CARD\",\"storePaymentCard\":true,\"adyenEncryptedData\":\"adyenjs_0_1_7$dTjmY0yWf5EXWtz8WfekCZ9I4uusWFSUNfQfrYMezqGVSVW7iS0VGWQ5RnxGcTKD2b2PLuftLS+fE2wLEZeb8IRRqiX8jQiWjUsp4Q07/GKqNiUv0pj7KSEI2bLG7z39fpTpEgZ7K863WDNqW08iJiCNr5qcUH77MIqE1p08tN3blBmHZn1fhzjAsNcr/9yYcxWvZi/crgDIPQWGAGjoK+v2kyQZ67EnCbbzxWZicsZUIvAL8lSixVhvunRvgF+GoMz6s5Xjhexb4+bO1T2vcvjftAiygEIRisakpKF9mOeKfJW76+OAWCgRIL+ayl8/8p10X+o+2b7XD7faqIoAcQ==$PIi42wyJt/5xxbLqsHXSfURQx5f1iuoDxmEymqupxrkmDiuwCGyzHTQs8mKL/IrLbvaUv9fqk4ZspvnzJ7pZ26tOKTIU5pjRV3Fl95+ztJKG9j0Qn+F8WaTUt6h7ycWPfylPX00qVmJnQu6oD7JydgQic4MTGUS9dxz0KhrW0FwNeEh4b5A+UhIGNU/WUEq7MPGNEhm/1UxtAWNNanqCQmwskq95ItpCMNyC3WORWMBzzaOosEZMbJyDODyqWq4Drol55+ZlOK5d5+7gH52++26EDAtsK0f8\",\"adyenEncryptedGenerationTime\":\"2017-01-16T13:27:41.915Z\",\"discountCodes\":[]},\"customer\":{\"emailAddress\":\"video@wahanda.com\",\"newsletterSignup\":\"true\",\"gift-option\":\"yes\",\"name\":\"etwtwtwe\",\"phoneNumber\":\"02086400366\",\"guests\":[{\"name\":\"etwtwtwe\"}]},\"delivery\":{\"address\":{}}}";
```
Comment (gregbeech) [inline src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:35]: Encoded JSON in a string is pretty gross 😢

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        String r = checkoutApi.checkout(body);
+        Assert.assertTrue(r.contains("error"));
+        System.out.println(r);
```
Comment (gregbeech) [inline src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:38]: Debugging code should be removed.

**Review by gregbeech — CHANGES_REQUESTED** [review]
Looks like a good start, but still some things that could be improved 👍

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java (hunk 7 above)
Comment (aguairon) [reply src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:35]: This is just to introduce it. I will flesh out and setting things properly (eg. json) later.

File: src/test/java/api/SiteApi.java (hunk 4 above)
Comment (aguairon) [reply src/test/java/api/SiteApi.java:21]: That is the way Intellij is insisting it should be formatted so I am going to leave as is to make my live easier

File: src/test/java/tests/search/checkout/CheckoutSubmissionTest.java (hunk 5 above)
Comment (aguairon) [reply src/test/java/tests/search/checkout/CheckoutSubmissionTest.java:31]: That is really on in preparation of the future tests

**Review by gregbeech — APPROVED** [review]

**Comment by gregbeech** [comment]
@aguairon looks good, :shipit:

**Pull request merged**